classloader-leak-prevention/classloader-leak-prevention-core/src/main/java/se/jiderhamn/classloader/leak/prevention/cleanup/JDK8151486CleanUp.java

```diff
@@ -0,0 +1,25 @@
+package se.jiderhamn.classloader.leak.prevention.cleanup;
+
+import se.jiderhamn.classloader.leak.prevention.ClassLoaderLeakPreventor;
+import se.jiderhamn.classloader.leak.prevention.ClassLoaderPreMortemCleanUp;
+
+import java.lang.reflect.Field;
+import java.util.Set;
+
+/**
+ * Clear the "domains" field of the parent ClassLoader.
+ *
+ * See <a href="https://bugs.openjdk.java.net/browse/JDK-8151486">JDK-8151486</a>
+ */
+public class JDK8151486CleanUp implements ClassLoaderPreMortemCleanUp {
+    @Override
+    public void cleanUp(ClassLoaderLeakPreventor preventor) {
+        Field field = preventor.findField(ClassLoader.class, "domains");
+        if (field != null) {
+            Set<?> domains = preventor.getFieldValue(field, preventor.getLeakSafeClassLoader());
```

Comment:
That might be a more general approach. In any case it doesn't hurt to clear the domains since they're not used for anything.